applications/UmatApplication/CMakeLists.txt

```diff
@@ -0,0 +1,60 @@
+set(CMAKE_INCLUDE_CURRENT_DIR ON)
+
+message("**** configuring KratosUmatApplication ****")
+
+include_directories( ${CMAKE_SOURCE_DIR}/kratos )
+include_directories( ${CMAKE_CURRENT_SOURCE_DIR}/../ConstitutiveModelsApplication )
+
+if(NOT (${CONSTITUTIVE_MODELS_APPLICATION} MATCHES "ON"))
+    message(SEND_ERROR "The ConsitutiveModels Application must be compiled before this one. Set CONSTITUTIVE_MODELS_APPLICATION=ON in the configure file.")
+endif(NOT (${CONSTITUTIVE_MODELS_APPLICATION} MATCHES "ON"))
+
+# generate variables with the sources
+set( KRATOS_UMAT_APPLICATION_SOURCES
+	${CMAKE_CURRENT_SOURCE_DIR}/umat_application.cpp
+        ${CMAKE_CURRENT_SOURCE_DIR}/umat_application_variables.cpp
+	${CMAKE_CURRENT_SOURCE_DIR}/custom_python/umat_python_application.cpp
+	${CMAKE_CURRENT_SOURCE_DIR}/custom_python/add_custom_constitutive_laws_to_python.cpp
+	${CMAKE_CURRENT_SOURCE_DIR}/custom_laws/umat_3D_law.cpp
+
+	#/${CMAKE_CURRENT_SOURCE_DIR}/custom_laws/small_strain_umat_3D_law.cpp
+	#${CMAKE_CURRENT_SOURCE_DIR}/custom_laws/large_strain_umat_3D_law.cpp
+
+	${CMAKE_CURRENT_SOURCE_DIR}/umat_models/mises_umat.f
+	#${CMAKE_CURRENT_SOURCE_DIR}/umat_models/umat_hypo.f
+	${CMAKE_CURRENT_SOURCE_DIR}/umat_models/hypoplastic_umat.f
+	${CMAKE_CURRENT_SOURCE_DIR}/umat_models/umat_fortran_wrapper.f
+	${CMAKE_CURRENT_SOURCE_DIR}/umat_models/xit.f
+	
+	${CMAKE_CURRENT_SOURCE_DIR}/custom_models/small_strain_umat_model.cpp
+	${CMAKE_CURRENT_SOURCE_DIR}/custom_models/von_mises_umat_small_strain_model.cpp
+	${CMAKE_CURRENT_SOURCE_DIR}/custom_models/hypoplastic_umat_small_strain_model.cpp
+	${CMAKE_CURRENT_SOURCE_DIR}/custom_models/large_strain_umat_model.cpp
+	${CMAKE_CURRENT_SOURCE_DIR}/custom_models/von_mises_umat_large_strain_model.cpp
+
+)
+
+# define library Kratos which defines the basic python interface
+add_library(KratosUmatApplication SHARED ${KRATOS_UMAT_APPLICATION_SOURCES})
+target_link_libraries(KratosUmatApplication KratosCore KratosConstitutiveModelsApplication)
+set_target_properties(KratosUmatApplication PROPERTIES PREFIX "")
```

Comment:
UMAT seems to be using KRATOS_API(UMAT_APPLICATION) macro but UMAT_APPLICATION in not defined here. This will cause it will never export in windows. Don't know if is intended

Reply:
You explined to me that there were problems with the fortran compilation in windows. If they are solve I will thank you if add the mentioned correction.

Reply:
You explained to me that there was some problems in fortran compilation for  windows. If they are solved I will thank you if you add the proper corrections.

Reply:
Hi, it is sort of intended. In this application we are using Fortran code, we thought it will be easier to compile in Linux and that's why we created this tiny application. Of course, this is almost a draft of the feature (be able to incorporate constitutive laws written in the Umat format, typical of comercial FEM softwares) that works but some more work is required to make it relevant.

Reply:
Well is not a problem per-se. It works with intel we just don't have the licence so we cannot use it, apart from that its just fine. 

In my opinion, it doesn't hurt to add the define as you done the hard part in the code, but I understand what you mean.